newdle/client/src/components/creation/FinalStep.js

```diff
@@ -132,35 +152,51 @@ export default function FinalStep() {
           color="violet"
           type="submit"
           disabled={!canSubmit}
-          onClick={createNewdle}
+          onClick={isEditing ? editNewdle : createNewdle}
           loading={submitting}
         >
-          <Trans>Create your newdle!</Trans>{' '}
-          <span role="img" aria-label="Newdle">
-            🍜
-          </span>
+          {!isEditing ? (
+            <>
+              <Trans>Create your newdle!</Trans>{' '}
+              <span role="img" aria-label="Newdle">
+                🍜
+              </span>
```

Comment:
Unrelated to your changes, but I think this is one of the cases where disabling the a11y warning instead of wrapping the emoji in a span with a label may be more appropriate; It's just decoration, a screen reader user would not have any benefit from it telling them something like "Create your newdle! newdle"...